### PR TITLE
Forward port and protocol correctly

### PR DESCRIPTION
**Pull request by oterral**

### CODE REVIEW

File: nginx/nginx.conf.in [hunk 1]
```diff
@@ -35,10 +35,10 @@ http {
     }
     location / {
       proxy_pass                          http://localhost:8135;
-      proxy_set_header Host               $host;
+      proxy_set_header Host               $host:$server_port;
```
Comment (danduk82) [inline nginx/nginx.conf.in]: why do you need the port in header Host? Hostname shouldn't have ports in it

Comment (oterral) [reply nginx/nginx.conf.in]: Otherwise when you're in local the style as bad link inside:
http://localhost:8134/styles/psychedelic/style.json

Comment (oterral) [reply nginx/nginx.conf.in]: Before the PR links instide the style were: 

https://localhost/styles....

Now with the pr it's:

http://localhost:8134/styles

Comment (oterral) [reply nginx/nginx.conf.in]: If you have a better idea. I take it.

Comment (danduk82) [reply nginx/nginx.conf.in]: go ahead...

**Comment by danduk82** [comment]
I was thinking that we will probably have to modify tileserver-gl because if we use cloudfront we will have to use another header instead of `Host` to map with the external (cloudfront) header. For example, with WMS we use `X-Forwarded-Host` for this. It is much easier to use several levels of backend caching if we would do the same here.

**Pull request merged**